crates/solidity/inputs/language/bindings/rules.msgb

```diff
@@ -797,54 +1081,82 @@ inherit .lexical_scope
     | [FixedKeyword]
     | [UfixedKeyword]
 )] {
-  let @elementary.symbol = (format "%{}" (source-text @keyword))
+  var symbol = (source-text @keyword)
```

Comment:
Q: since these involve string matching and formatting, should we have specific queries instead?

- dedicated queries for `uint`, `int`, `ufixed`, `fixed` that each check for the single value.
- another general query for the others that don't need string matching.

Reply:
Can we do this without enumerating all the possible values in the general query? That will result in string matching anyway, just in a different place. I don't see how it's possible otherwise, given that we don't have a negation operator for queries.

Aside: there are a couple of cases where having a negation query operator would have been useful. I remember @AntonyBlakey asking me about that, and at the time I didn't have any specific use cases. But I've now seen a handful, this one being one of them.

Reply:
A negation operator would be easy, as you probably know.

Reply:
Yeah, the semantics especially in combination with other operators can be hard to design.

Reply:
Oh, I now see what you mean. That looks reasonable. I'll change that.